codeguru_profiler_agent/sdk_reporter/sdk_reporter.py

```diff
@@ -80,7 +96,13 @@ def report(self, profile):
 
         :param profile: Profile to be encoded and reported to the profiler backend service.
         :return: True if profile gets reported successfully; False otherwise.
+
+        Note:
+        For an agent running on AWS Lambda, if the environment variables for Profiling using
+        Lambda layers are set, it tries to create a Profiling Group whenever a ResourceNotFoundException
+        is encountered.
         """
+        global is_create_pg_called_during_submit_profile
```

Comment:
Why are you defining it global here? Can't you do that at line #22?

Reply:
removed global altogether. was using a class variable so global was not needed!